Src/Plugins/Security/SEC101_047.MySqlConnectionStringValidator.cs

```diff
@@ -62,20 +59,32 @@ public static string IsValidDynamic(ref string fingerprint, ref string message)
                                                       ref string fingerprintText,
                                                       ref string message)
         {
-            string host = ParseExpression(RegexEngine, matchedPattern, HostRegex);
-            string account = ParseExpression(RegexEngine, matchedPattern, AccountRegex);
-            string password = ParseExpression(RegexEngine, matchedPattern, PasswordRegex);
-            string database = ParseExpression(RegexEngine, matchedPattern, DatabaseRegex);
-            string port = ParseExpression(RegexEngine, matchedPattern, PortRegex);
-
-            if (string.IsNullOrWhiteSpace(host) ||
-                string.IsNullOrWhiteSpace(database) ||
-                string.IsNullOrWhiteSpace(account) ||
-                string.IsNullOrWhiteSpace(password))
+            if (!groups.TryGetNonEmptyValue("account", out string account) ||
+                !groups.TryGetNonEmptyValue("password", out string password))
             {
                 return nameof(ValidationState.NoMatch);
             }
 
+            // Our matches can sometimes fail to find a host (due to it being constructed in code)
+            // However, the credentials can still be valid, so we should return "unknown".
+            // Grab the empty host here and then short circuit in dynamic validation.
+
+            Fingerprint fingerprint = new Fingerprint()
+            {
+                Account = account,
+                Password = password,
+            };
+
+            if (!groups.TryGetNonEmptyValue("host", out string host))
```

Comment:
>groups.TryGetNonEmptyValue("host", out string host) [](start = 17, length = 51)

this is strange. if we  want to do this, i think its better to create a new rule just to grab account and password